pkg/controller/deployments/controller.go

```diff
@@ -68,10 +69,23 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 		log.Info("failed to get deployment in namespace " + request.Namespace + " with name  " + d.Name)
 		return reconcile.Result{}, nil
 	}
-	if _, ok := d.Labels[domain.HeimdallMonitored]; ! ok{
+	should, err := validation.ShouldCheck(d)
+	if err != nil{
+		if validation.IsParseErr(err){
+			delete(d.Annotations, domain.HeimdallLastChecked)
+			if err := r.client.Update(context.TODO(),d); err != nil{
+				// in this case we will requeue log the error and requeue to ensure we dont keep retrying the checks
+				log.Error(err, " failed to label deployment " + request.Namespace + " " + request.Name)
+				return reconcile.Result{RequeueAfter: requeAfterFourHours}, nil
+			}
+			return reconcile.Result{}, nil
+		}
+	}
+	if ! should{
+		log.Info("critera for re checking not met")
```

Comment:
@maleck13 I guess you are implementing this? https://issues.jboss.org/browse/INTLY-4072

If you are i guess we can close this PR and verify when you are done. I'll delete that and put your ticket back to in progress.